### PR TITLE
added support to a generic stm32f407vet6 with arduino/stm32duino

### PR DESCRIPTION
**Pull request by sczekajewski**
Tests done with Black STM32F4xx board (http://wiki.stm32duino.com/index.php?title=STM32F407).
Upload and debug both require stlink connected.

### CODE REVIEW

File: boards/genericSTM32F407VET6.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{
+  "build": {
+    "core": "stm32",
+    "cpu": "cortex-m4",
+    "extra_flags": "-mthumb -DSTM32F4 -DSTM32_HIGH_DENSITY",
```
Comment (ubis) [inline boards/genericSTM32F407VET6.json]: Please remove `-mthumb ` option, it is set in build script already.

File: boards/genericSTM32F407VET6.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          "executable": "bin/openocd",
+          "package": "tool-openocd"
+        }
+      }
+    }
+  },
+  "frameworks": [
+    "arduino"
+  ],
+  "name": "STM32GenericF407VET6",
```
Comment (ubis) [inline boards/genericSTM32F407VET6.json]: Please rename board to `STM32F407VE (192k RAM. 512k Flash)`

File: builder/frameworks/arduino/maple/stm32f4.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    CXXFLAGS=[
+        "-mthumb",
+        "-MMD",
+        "-std=gnu++11",
+        "-nostdlib",
+        "-fno-rtti",
+        "-fno-exceptions",
+        "--param",
+        "max-inline-insns-single=500"],
```
Comment (ubis) [inline builder/frameworks/arduino/maple/stm32f4.py]: For the same flags in C and C++, use `CCFLAGS`, here's an [example](https://github.com/platformio/platform-ststm32/blob/develop/builder/frameworks/arduino/maple/stm32f1.py#L101).
And flags like `mthumb`; `nostdlib`; `fno-rtti`; `fno-exceptions` are already defined in [build script](https://github.com/platformio/platform-ststm32/blob/develop/builder/main.py#L39). No need to include them in here.

File: builder/frameworks/arduino/maple/stm32f4.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+    CPPDEFINES=[
+        ("DEBUG_LEVEL", "DEBUG_NONE"),
+        ("BOARD_%s" % variant),
+        ("ERROR_LED_PORT", error_led_port),
+        ("ERROR_LED_PIN", error_led_pin),
+        ("ARDUINO", 10610),
+        ("ARDUINO_%s" % variant.upper()
+            if not "nucleo" in board.id else "STM_NUCLEO_F103RB"),
```
Comment (ubis) [inline builder/frameworks/arduino/maple/stm32f4.py]: Please remove if statement, since there's no F103RB board in here.

**Comment by ubis** [comment]
Hi, thanks for PR!
I quickly reviewed your PR and you should take a look of my suggestions.

**Comment by sczekajewski** [comment]
Modified scripts according to your comments.

**Comment by ubis** [comment]
I think this can be merged. Later we will do some changes to add more frameworks/upload protocols.
@ivankravets WDYT?

**Comment by ivankravets** [comment]
@sczekajewski did you test debugging? Does it work?

**Comment by sczekajewski** [comment]
it requires you to place a line in platformio.ini like: 
``debug_tool = stlink
And then you can debug your code (at least I could on my Black STM32F4xx board)

**Comment by ivankravets** [comment]
Thanks!

**Pull request merged**